### PR TITLE
Refactor FilterByValueBlock to use BaseBlock and new directory structure

### PR DESCRIPTION
**Pull request by eshwarprasadS**
## Summary

- Migrates `FilterByValueBlock` from monolithic `utilblocks.py` to new directory structure: `src/sdg_hub/blocks/filtering/filter_by_value.py`
- Updates the block to inherit from `BaseBlock` with standardized constructor patterns
- Adds explicit `input_cols` and `output_cols` parameters while maintaining full backward compatibility with existing flows
- Implements smart column validation that automatically includes `filter_column` in `input_cols`

## Key Changes

- **New location**: `src/sdg_hub/blocks/filtering/filter_by_value.py` 
- **BaseBlock integration**: Now uses Rich logging panels, automatic validation, and standardized error handling
- **Backward compatibility**: All existing YAML flows continue to work without modifications
- **Enhanced constructor**: Supports both legacy style and new BaseBlock pattern with explicit column specifications
- **Comprehensive tests**: Added tests for BaseBlock integration, edge cases, and multiple input column scenarios

## Backward Compatibility

✅ All existing flows using `FilterByValueBlock` continue to work unchanged  
✅ Import compatibility maintained through updated `__init__.py` files  
✅ Registry registration preserved with same block name  
✅ All constructor parameters work identically  

## Test Plan

- [x] All existing tests pass (11/11)
- [x] New BaseBlock-style constructor tests added
- [x] Import compatibility verified
- [x] Real flow configurations tested successfully
- [x] Edge cases covered (multiple input_cols, missing filter_column)

Addresses #224 and follows the filtering blocks reorganization outlined in #223.

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/sdg_hub/blocks/filtering/filter_by_value.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+from typing import Any, Callable, Dict, List, Optional, Type, Union
+import operator
+
+# Third Party
+from datasets import Dataset
+
+# Local
+from ...logger_config import setup_logger
+from ...registry import BlockRegistry
+from ..base import BaseBlock
```
[the remark this answers is not in the document]
Comment (eshwarprasadS) [reply src/sdg_hub/blocks/filtering/filter_by_value.py:18]: This is not applicable here because this PR is about migrating `FilterByValueBlock` to use `BaseBlock` (the new ABC for creating blocks). The old conventions are inapplicable to this PR
